app/scripts/modules/core/src/search/infrastructure/RecentlyViewedItems.tsx

```diff
@@ -0,0 +1,76 @@
+import { IPromise } from 'angular';
+import * as React from 'react';
+import { BindAll } from 'lodash-decorators';
+import { Observable } from 'rxjs/Observable';
+import { Subject } from 'rxjs/Subject';
+
+import { IRecentHistoryEntry } from 'core/history';
+import { ReactInjector } from 'core/reactShims';
+
+import { ISearchResult, ISearchResultPodData, SearchResultPods } from './SearchResultPods';
+
+export interface IRecentlyViewedItemsState {
+  recentItems: ISearchResultPodData[];
+}
+
+@BindAll()
+export class RecentlyViewedItems extends React.Component<{}, IRecentlyViewedItemsState> {
+  public state: IRecentlyViewedItemsState = { recentItems: [] };
+  private categories = ['projects', 'applications', 'loadBalancers', 'serverGroups', 'instances', 'securityGroups'];
+  private recentHistoryService = ReactInjector.recentHistoryService;
+  private search = ReactInjector.infrastructureSearchService.getSearcher();
+
+  private refresh$ = new Subject<string[]>();
+
+  public componentWillUnmount() {
+    this.refresh$.complete();
+  }
+
+  private updateRecentItems() {
+    this.refresh$.next(this.categories);
+  }
+
+  public componentDidMount() {
+    this.refresh$.switchMap((categories: string[]) => {
+      return Observable.forkJoin(categories.map(category => {
+        const config = this.search.getCategoryConfig(category);
+        const items = this.recentHistoryService.getItems(category);
+        const promises = items.map(item => this.getFullHistoryEntry(category, item));
+        return Promise.all(promises).then(results => ({ category, config, results }));
+      }));
+    }).map(recentItems => {
+      return recentItems.filter(item => item.results.length)
+    }).subscribe(recentItems => {
+      this.setState({ recentItems })
+    });
+
+    this.updateRecentItems();
+  }
+
+  /** fetches the displayName and adds it to the history entry*/
```

Comment:
odd spacing around this comment (two `**` at the start, no space between last word and `*`)

Reply:
FYI I use two stars because (like in java) the two ** at the start denotes a method description comment (javadoc format/jsdoc format)